### PR TITLE
ASP-2438 Speed up tests on Jobbergate API

### PR DESCRIPTION
**Pull request by fschuch**
#### What
As a Jobbergate developer, I would like to have a quicker feedback loop when running the tests on the Jobbergate API.

#### Why
Increase maintainability and speedup any future task on the Jobbergate API.

`Task`: https://jira.scania.com/browse/ASP-2438

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: .github/workflows/test_on_push.yaml [hunk 1]
```diff
@@ -41,25 +42,25 @@ jobs:
           TEST_DATABASE_USER: test
           TEST_DATABASE_PSWD: test-pswd
           TEST_DATABASE_NAME: test-jobbergate
-          POETRY_HTTP_BASIC_PYPICLOUD_USERNAME: admin
-          POETRY_HTTP_BASIC_PYPICLOUD_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-        run: make qa
+        run: |
+          poetry env use "3.8"
```
Comment (fschuch) [inline .github/workflows/test_on_push.yaml:46]: Poetry was using the Python version it was installed from, the one from the system. Here it is forced to use the one installed by `actions/setup-python`, like suggested by: https://github.com/actions/setup-python/issues/374#issuecomment-1088938718

File: .github/workflows/test_on_push.yaml [hunk 2]
```diff
@@ -41,25 +42,25 @@ jobs:
...
+          cache: "poetry"
+          cache-dependency-path: jobbergate-cli/poetry.lock
       - name: "run quality control checks"
         working-directory: jobbergate-cli
-        env:
-          POETRY_HTTP_BASIC_PYPICLOUD_USERNAME: admin
-          POETRY_HTTP_BASIC_PYPICLOUD_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-        run: make qa
+        run: |
+          poetry env use "3.6"
```
Comment (fschuch) [inline .github/workflows/test_on_push.yaml:65]: see https://github.com/omnivector-solutions/jobbergate/pull/212/files#r1061435455

File: jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py [hunk 3]
```diff
@@ -14,6 +14,8 @@
 from jobbergate_api.apps.permissions import Permissions
 from jobbergate_api.storage import database, fetch_instance
 
+pytestmark = pytest.mark.usefixtures("startup_event_force")
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py:17]: Please add a comment for these that explains why this is here.

**Review by dusktreader — CHANGES_REQUESTED** [review]

File: jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py (hunk 3 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py:19]: solved in [2c6e84c](https://github.com/omnivector-solutions/jobbergate/pull/212/commits/2c6e84cb5081f4457c9468553293adeea73d618f)

**Review by dusktreader — APPROVED** [review]

**Pull request merged**